### PR TITLE
rb_yjit_constant_ic_update

### PR DESCRIPTION
**Pull request by kddnewton**

### CODE REVIEW

**Comment by noahgibbs** [comment]
LGTM, but I worked with you on it ;-)

**Comment by noahgibbs** [comment]
Looks like something is panicking, though: https://github.com/Shopify/ruby/runs/5630779426?check_suite_focus=true

**Comment by maximecb** [comment]
Looks like the failing test is related to this PR

File: yjit/src/invariants.rs [hunk 1]
```diff
@@ -289,66 +289,52 @@ pub fn block_assumptions_free(block: BlockRef) {
...
 
-    RB_VM_LOCK_ENTER();
-    rb_vm_barrier(); // Stop other ractors since we are going to patch machine code.
-    {
-        const struct rb_iseq_constant_body *const body = iseq->body;
-        VALUE *code = body->iseq_encoded;
-        const unsigned get_insn_idx = ic->get_insn_idx;
+    with_vm_lock(src_loc!(), || {
+        // Stop other ractors since we are going to patch machine code.
+        unsafe { rb_vm_barrier() };
```
Comment (XrXr) [inline yjit/src/invariants.rs]: `with_vm_lock()` does this for you so you don't need to do this.

**Comment by maximecb** [comment]
@kddnewton let us know if you need some help

**Pull request merged**